eng/pipelines/templates/steps/template-test-run-job.yml

```diff
@@ -43,27 +48,100 @@ steps:
     Write-Host "##vso[task.setvariable variable=CleanupImmediate]$CleanupImmediate"
   displayName: Set test run parameters
 
-- task: Bash@3
-  displayName: Test templates
-  env:             
-    # Required secrets for Terraform service principal authentication
-    ARM_CLIENT_ID: $(arm-client-id)
-    ARM_CLIENT_SECRET: $(arm-client-secret)
-    ARM_TENANT_ID: $(arm-tenant-id)
+- pwsh: |
+    git config --global user.name "azure-sdk"
+    git config --global user.email "azuresdk@microsoft.com"
+    echo "https://azure-sdk:$(azuresdk-github-pat)@github.com" > ~/.git-credentials
+    git config --global credential.helper "store"
+    Write-Host "##vso[task.setvariable variable=folderPath;]$pwd"
```

Comment:
We don't need to use `folderPath` here. The default location of all commands is `$(Build.SourcesDirectory)` https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml

Naturally there are some caveats, but in this case since the pipeline is only checking out a [single repo](https://github.com/hemarina/azure-dev/blob/fix-504/eng/pipelines/templates/jobs/run-template-tests.yml#L68-L69) so the default working directory is the root of the `azure-dev` repo as it is checked out on the machine.